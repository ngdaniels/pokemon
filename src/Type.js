import React, { Component } from "react";
import styled from '@emotion/styled'

class Type extends Component {
    render() {
        let { type } = this.props;

        let Element = styled.button`
            background-color: ${props => props.background};
            color: white;
            border: 1px solid;
            border-radius: 20%/50%;
            margin: 5px;
            padding: 8px 10px;
            font-weight: bold;
            width: 100px;
        `

        let background;
        switch (type) {
            case "normal":
                background = "#a8a878";
                break;
            case "fighting":
                background = "#c03028";
                break;
            case "flying":
                background = "#a890f0";
                break;
            case "poison":
                background = "#a040a0";
                break;
            case "ground":
                background = "#e0c068";
                break;
            case "rock":
                background = "#b8a038";
                break;
            case "bug":
                background = "#a8b820";
                break;
            case "ghost":
                background = "#705898";
                break;
            case "steel":
                background = "#b8b8d0";
                break;
            case "fire":
                background = "#f08030";
                break;
            case "water":
                background = "#6890f0";
                break;
            case "grass":
                background = "#78c850";
                break;
            case "electric":
                background = "#f8d030";
                break;
            case "psychic":
                background = "#f85888";
                break;
            case "ice":
                background = "#98d8d8";
                break;
            case "dragon":
                background = "#7038f8";
                break;
            case "dark":
                background = "#705848";
                break;
            case "fairy":
                background = "#ee99ac";
                break;
            case "unknown":
                background = "#68a090";
                break;
            case "shadow":
                background = "#000000";
                break;
            default:
                break;
        }

        return (
            <Element background={background}>
                {type.toUpperCase()}
            </Element>
        )
    }
}

export default Type;