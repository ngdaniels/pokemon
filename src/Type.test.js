import { render } from '@testing-library/react';
import Type from './Type';

describe('Pokémon Type', () => {
	test('Pokémon Type undefined', () => {
		let {asFragment} = render(<Type/>);
        expect(asFragment()).toMatchSnapshot();
    });
    
    test('Pokémon Type defined', () => {
		let {asFragment} = render(<Type type="dragon"/>);
        expect(asFragment()).toMatchSnapshot();
    });
});