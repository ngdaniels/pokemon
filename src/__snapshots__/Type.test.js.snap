// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokémon Type Pokémon Type defined 1`] = `
<DocumentFragment>
  <button
    class="css-h8mfs4"
  >
    DRAGON
  </button>
</DocumentFragment>
`;

exports[`Pokémon Type Pokémon Type undefined 1`] = `
<DocumentFragment>
  <button
    class="css-8zi4cn"
  >
    ???
  </button>
</DocumentFragment>
`;
